variables.tf

```diff
@@ -12,6 +12,36 @@ variable "name" {
   description = "The name to give the IBM Event Notification instance created by this module."
 }
 
+variable "cos_destination_name" {
+  type        = string
+  description = "The name to give the IBM Cloud Object Storage destination which will be created for storage of failed delivery events."
+  default     = "COS Destination"
+}
+
+variable "cos_bucket_name" {
+  type        = string
+  description = "The name of an existing IBM Cloud Object Storage bucket which will be used for storage of failed delivery events."
+  default     = null
+}
+
+variable "cos_instance_id" {
+  type        = string
+  description = "The ID of the IBM Cloud Object Storage instance in which the bucket defined in the cos_bucket_name variable exists."
+  default     = null
+}
+
+variable "cos_region" {
+  type        = string
+  description = "The region in which the cos bucket is located."
+  default     = "us-south"
+}
+
+variable "cos_endpoint" {
+  type        = string
+  description = "The endpoint url for your cos instance."
```

Comment:
Mention its required if `cos_integration_enabled` is true.

Maybe also link to the COS endpoint doc https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-endpoints